sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/export/IntervalMetricReader.java

```diff
@@ -57,16 +64,30 @@ public static IntervalMetricReaderBuilder builder() {
     return new IntervalMetricReaderBuilder(InternalState.builder());
   }
 
-  @SuppressWarnings("FutureReturnValueIgnored")
   IntervalMetricReader(InternalState internalState) {
     this.exporter = new Exporter(internalState);
     this.scheduler =
         Executors.newScheduledThreadPool(1, new DaemonThreadFactory("IntervalMetricReader"));
-    this.scheduler.scheduleAtFixedRate(
-        exporter,
-        internalState.getExportIntervalMillis(),
-        internalState.getExportIntervalMillis(),
-        TimeUnit.MILLISECONDS);
+  }
+
+  /**
+   * Starts this {@link IntervalMetricReader} to report to the configured exporter.
+   *
+   * @return this for fluent usage along with the builder.
+   */
+  public IntervalMetricReader start() {
+    synchronized (lock) {
+      if (scheduledFuture != null) {
```

Comment:
Can you add a unit test for double-start?

Reply:
will do, good call.

Reply:
done

Reply:
hmm. I wrote a test that passed locally, and failed when we allowed more than one scheduled job. But, it fails on CI. Do you have some advice on how to write a test like this that asserts a negative state in a multi-threaded situation that will work on CI? :)

Reply:
Hmm - one idea is to record the export time in the test's `MetricExporter` - if any two consecutive exports have a delta less than the export interval. Since it's a single thread I think this invariant holds reliably.

Reply:
I'll give it a shot

Reply:
After many tries at various things (and learning that the scheduled executor timing is pretty darn sloppy at the millisecond level), I took the advice of @breedx-splk and passed in a mock executor service and just verified it was only called once.